File: 0139_Word Break/WordBreak_BruteForce_TLE.cs
public class Solution {
    public bool WordBreak(string s, IList<string> wordDict) {
        return CanBreak(s,  new HashSet<string>(wordDict));
        
    }
    
    private bool CanBreak(string s,  HashSet<string> set)
    {
        if(set.Contains(s))
        {
            return true;
        }
        
        for(int i=1;i<s.Length;i++)
        {
            if(set.Contains(s.Substring(i)) && CanBreak(s.Substring(0,i), set))
            {
                return true;
            }
        }
        
        return false;
    }
}

File: 0139_Word Break/WordBreak_BruteForce_TLE.java
class Solution {
    public boolean wordBreak(String s, List<String> wordDict) {
        Set<String> set = new HashSet<>(wordDict);
        return canBreak(s, set);
    }
    
    private boolean canBreak(String s, Set<String> set) {
        if(set.contains(s)) return true;
        for(int i=1;i<s.length();i++) {
            if(set.contains(s.substring(0, i)) && canBreak(s.substring(i), set)) return true;
        }
        
        return false;
    }
}